hibernate-core/src/main/java/org/hibernate/type/descriptor/java/JdbcTimeJavaTypeDescriptor.java

```diff
@@ -189,19 +203,29 @@ public boolean areEqual(Date one, Date another) {
 			return new Time( ( (Calendar) value ).getTimeInMillis() );
 		}
 
-		if ( value instanceof Date ) {
-			return new Time( ( (Date) value ).getTime() );
-		}
-
-		if ( value instanceof LocalTime ) {
-			return Time.valueOf( (LocalTime) value );
-		}
-
 		throw unknownWrap( value.getClass() );
 	}
 
 	@Override
 	public int getDefaultSqlPrecision(Dialect dialect, JdbcType jdbcType) {
-		return 0; //seconds (currently ignored since Dialects don't parameterize time type by precision)
+		//seconds (currently ignored since Dialects don't parameterize time type by precision)
+		return 0;
+	}
+
+
+	public static class TimeMutabilityPlan extends MutableMutabilityPlan<Date> {
+		public static final TimeMutabilityPlan INSTANCE = new TimeMutabilityPlan();
+		@Override
+		public Date deepCopyNotNull(Date value) {
+			if ( value == null ) {
+				return null;
+			}
+
+			if ( value instanceof Time ) {
+				return value;
```

Comment:
Should probably use this since it is a mutable type?

```suggestion
				return value.clone();
```

Reply:
yep, missed this.  though oddly did not change it either.